wordcloud_style.py

```diff
@@ -17,10 +17,24 @@ def select_colour(self, word, value, word_to_values):
 class QuantizedColourSelector(ColourSelector):
     """Apply colour based on a user provided list."""
 
+    def __init__(self, colour_list, max_value):
+        self.colour_list = colour_list
+        self.max_value = max_value
+
+    def on_begin_draw(self, word_to_values):
+        # Docstring from parent class:
+        """Handle any set up required before drawing the word cloud.
+
+        Called by a VisualWordCloud before the first word is drawn.
+        Override this method in a child class to do any set up.
+
+        :param word_to_values: Dictionary of words to their values."""
+        self.max_value = max(word_to_values.values())
+        return self.max_value
+
     def select_colour(self, word, value, word_to_values):
-        user_colours = [(255, 0, 0), (255, 127, 0), (255, 255, 0), (0, 255, 0), (0, 0, 255), (75, 0, 130),
-                   (148, 0, 211)]  # This is a list of RGB codes for the colours of the rainbow
-        max_value = max(word_to_values.values())
+
+        # max_value = max(word_to_values.values())
 
         # i = 1
         # for value in word_to_values.values():
```

Comment:
You're trying to loop over all words here. But select_colour is just selecting the colour for a single word. Perhaps you meant to iterate (loop) over colours instead?